v2: panic in inline table parsing

**Describe the bug**
Given `a={ `:
```
panic: runtime error: index out of range [0] with length 0
goroutine 1012 [running]:
runtime/debug.Stack()
        /home/x/sdk/gotip/src/runtime/debug/stack.go:24 +0x65
testing.tRunner.func1.2({0x65f2e0, 0xc0003fb8a8})
        /home/x/sdk/gotip/src/testing/testing.go:1281 +0x267
testing.tRunner.func1()
        /home/x/sdk/gotip/src/testing/testing.go:1288 +0x218
panic({0x65f2e0, 0xc0003fb8a8})
        /home/x/sdk/gotip/src/runtime/panic.go:1038 +0x215
github.com/pelletier/go-toml/v2.(*parser).parseInlineTable(0xc0001c9260, {0xc00017785a, 0x2, 0x6})
        /home/x/src/github.com/pelletier/go-toml/parser.go:357 +0x41a
github.com/pelletier/go-toml/v2.(*parser).parseVal(0xc0001c9260, {0xc00017785a, 0x2, 0x6})
        /home/x/src/github.com/pelletier/go-toml/parser.go:314 +0xa89
github.com/pelletier/go-toml/v2.(*parser).parseKeyval(0xc0001c9260, {0xc000177858, 0x4, 0x8})
        /home/x/src/github.com/pelletier/go-toml/parser.go:229 +0x2ee
github.com/pelletier/go-toml/v2.(*parser).parseExpression(0xc0001c9260, {0xc000177858, 0x84fdc0, 0x0})
        /home/x/src/github.com/pelletier/go-toml/parser.go:122 +0xad
github.com/pelletier/go-toml/v2.(*parser).NextExpression(0xc0001c9260)
        /home/x/src/github.com/pelletier/go-toml/parser.go:61 +0x119
github.com/pelletier/go-toml/v2.(*decoder).nextExpr(...)
        /home/x/src/github.com/pelletier/go-toml/unmarshaler.go:143
github.com/pelletier/go-toml/v2.(*decoder).fromParser(0xc0000ab6d0, {0x62f040, 0xc0001a18c0, 0xc000191a70})
        /home/x/src/github.com/pelletier/go-toml/unmarshaler.go:191 +0x65
github.com/pelletier/go-toml/v2.(*decoder).FromParser(0xc0000ab6d0, {0x61eb40, 0xc0001a18c0})
        /home/x/src/github.com/pelletier/go-toml/unmarshaler.go:177 +0xf8
github.com/pelletier/go-toml/v2.Unmarshal({0xc000177858, 0x4, 0x8}, {0x61eb40, 0xc0001a18c0})
        /home/x/src/github.com/pelletier/go-toml/unmarshaler.go:27 +0x112
```

**To Reproduce**
https://play.golang.org/p/GfYx2jgTweq
```go
toml.Unmarshal([]byte(`a={ `), &v)
```

**Expected behavior**
No panic.

**Versions**
 - go-toml: v2 4a5ae9e81e09e3cc5197fccde9105e9fa5ff65b9
 - go version devel go1.17-9c56a64673 Fri Sep 3 18:34:56 2021 +0000 linux/amd64

Found with Go dev.fuzz.

--- a/parser.go
+++ b/parser.go
@@ -353,7 +353,13 @@ func (p *parser) parseInlineTable(b []byte) (ast.Reference, []byte, error) {
 	var err error
 
 	for len(b) > 0 {
+		previousB := b
 		b = p.parseWhitespace(b)
+
+		if len(b) == 0 {
+			return parent, nil, newDecodeError(previousB[:1], "inline table is incomplete")
+		}
+
 		if b[0] == '}' {
 			break
 		}

--- a/unmarshaler_test.go
+++ b/unmarshaler_test.go
@@ -1734,6 +1734,12 @@ func TestIssue581(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestIssue586(t *testing.T) {
+	var v interface{}
+	err := toml.Unmarshal([]byte(`a={ `), &v)
+	require.Error(t, err)
+}
+
 //nolint:funlen
 func TestUnmarshalDecodeErrors(t *testing.T) {
 	examples := []struct {

